src/main/java/com/conveyal/datatools/manager/jobs/PeliasUpdateJob.java

```diff
@@ -0,0 +1,243 @@
+import org.apache.http.Header;
+import org.apache.http.HttpResponse;
+import org.apache.http.message.BasicHeader;
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.stream.Collectors;
+
+import static com.mongodb.client.model.Filters.in;
+
+public class PeliasUpdateJob extends MonitorableJob {
+    /**
+     * The deployment to send to Pelias
+     */
+    private Deployment deployment;
+
+    /**
+     * The workerId our request has on the webhook server. Used to get status updates
+     */
+    private String workerId;
+
+
+    /**
+     * Timer used to poll the status endpoint
+     */
+    private Timer timer;
+
+    /**
+     * Webhook authorization from username and password
+     */
+    private Header webhookAuthorization;
+
+    /**
+     * S3 URI to upload logs to
+     */
+    private AmazonS3URI logUploadS3URI;
+
+    public PeliasUpdateJob(Auth0UserProfile owner, String name, Deployment deployment, AmazonS3URI logUploadS3URI) {
+        super(owner, name, JobType.UPDATE_PELIAS);
+        this.deployment = deployment;
+        this.timer = new Timer();
+        this.logUploadS3URI = logUploadS3URI;
+    }
+
+    /**
+     * This method must be overridden by subclasses to perform the core steps of the job.
+     */
+    @Override
+    public void jobLogic() throws Exception {
+        status.message = "Launching custom geocoder update request";
+        workerId = this.makeWebhookRequest();
+        status.percentComplete = 1.0;
+
+        // Give server 1 second to create worker
+        Thread.sleep(1000);
+        // Check status every 2 seconds
+        timer.schedule(new StatusChecker(), 0, 2000);
+
+    }
+
+    private void getWebhookStatus() {
+        URI url = getWebhookURI(deployment.peliasWebhookUrl + "/status/" + workerId);
+
+        List<Header> headers = new ArrayList<>();
+        headers.add(webhookAuthorization);
+
+        HttpResponse response = HttpUtils.httpRequestRawResponse(url, 1000, HttpUtils.REQUEST_METHOD.GET, null, headers);
+
+        // Convert raw body to JSON
+        String jsonResponse;
+        try {
+            jsonResponse = EntityUtils.toString(response.getEntity());
+        }
+        catch (NullPointerException | IOException ex) {
+            status.fail("Webhook status did not provide a response!", ex);
+            timer.cancel();
+            return;
+        }
+
+        // Parse JSON
+        PeliasWebhookStatusMessage statusResponse = null;
+        try {
+            statusResponse = JsonUtil.objectMapper.readValue(jsonResponse, PeliasWebhookStatusMessage.class);
+        } catch (IOException ex) {
+            status.fail("Server refused to provide a valid status update. Are the credentials correct?", ex);
+            timer.cancel();
+            return;
+        }
```

Comment:
I'd be a little more lenient with these responses. Maybe give them up to 3 times to be ok in case there is something weird that happens for a minute or two and then everything is ok again.

Reply:
Good point! Added a counter in 3ed943abf53df3baa41486c2d6efee5f3ca257b3